src/transaction/txParser.ts

```diff
@@ -183,26 +237,47 @@ const parseTxCerts = (txCertificates: any[]): _Certificate[] => {
   )
 }
 
-const parseTxWithdrawals = (
-  withdrawals: any,
-): _Withdrawal[] => {
+const parseTxWithdrawals = (withdrawals: any): _Withdrawal[] => {
   if (!isWithdrawalsMap(withdrawals)) {
     throw Error(Errors.WithrawalsParseError)
   }
   return Array.from(withdrawals).map(([address, coins]): _Withdrawal => ({ address, coins: BigInt(coins) }))
 }
 
-const parseTtl = (ttl: any): number => {
+const parseFee = (fee: any): BigInt => {
+  if (!isLovelace(fee)) {
+    throw Error(Errors.FeeParseError)
+  }
+  return BigInt(fee)
+}
+
+const parseTtl = (ttl: any): number | undefined => {
   if (ttl === null || ttl === undefined) {
-    throw Error(Errors.MssingTTLParseError)
+    return undefined
+  }
+  if (!Number.isInteger(ttl)) {
+    throw Error(Errors.TTLParseError)
   }
   return ttl
 }
 
+const parseValidityIntervalStart = (validityIntervalStart: any): number | undefined => {
+  if (validityIntervalStart === null || validityIntervalStart === undefined) {
+    return undefined
+  }
+  if (typeof validityIntervalStart !== 'number') {
+    throw Error(Errors.ValidityIntervalStartParseError)
+  }
+  return validityIntervalStart
```

Comment:
Same as with the ttl, also Number.isInterger

Reply:
Refactored